File: 0201-0300/0219-Contains Duplicate II/0219-Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> table;
        for (int i = 0; i < nums.size(); ++i) {
            if (table.find(nums[i]) != table.end() && i - table[nums[i]] <= k) {
                return true;
            }
            else {
                table[nums[i]] = i;
            }
        }
        
        return false;
    }
};

File: 0201-0300/0219-Contains Duplicate II/0219-Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        table = {}
        for i, num in enumerate(nums):
            if num in table and i - table[num] <= k:
                return True
            else:
                table[num] = i
        return False
